friends/src/Components/FriendList.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import styled from 'styled-components';
+const FriendList = (props) => {
+	return (
+		<div>
+			{props.friends.map((friend) => (
+				<ul key={friend.id}>
+					<button type="submit" onClick={(props) => props.updateFriend(friend.id)}>
```

Comment:
This is where the `App.js` function is called, since it's sent here as props. The trouble is, you're sending the id, but not the friend object.

You'll need to find a way to marry the two pieces of logic from `FriendList` and `FriendForm`. I think the simplest way to do this would be to render an update form below each friend, and send `friend.id` and the info from your form handler to `props.updateFriend`. If you want to get extra fancy, you can create some functionality similar to `React-Router-Movies`, where you click on a friend and it shows an update form for that friend.